### PR TITLE
Feat/appl records/v75

### PR DESCRIPTION
**Pull request by victorjulien**
Continuation of #6713.

Rebased to master.
Frame logging now disabled by default.
Rust no longer exposes the C pointer as this didn't work well.
**TODO** Rust API may need more cleanup now that we only work on an ID.
Adds a very basic Telnet app-layer parser to demonstrate what a simple frame only parser could look like.
Frame keyword now can use a shorter notation:
`alert smb .... frame:smb1.pdu`, so leaving out the app proto from the frame type if it was already specified.
TLS frames improved. SSLv2 added.

suricata-verify-pr: 630

### CODE REVIEW

File: suricata.yaml.in [hunk 1]
```diff
@@ -168,7 +168,7 @@ outputs:
         # app layer frames
         - frame:
             # disabled by default as this is very verbose.
-            enabled: no
+            enabled: yes
```
Comment (victorjulien) [inline suricata.yaml.in:171]: TODO undo

File: rust/src/telnet/parser.rs [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+    let (i, _) = tag(b"\xff")(oi)?;
+    let (i, cmd) = le_u8(i)?;
+    let (i, d) = match cmd {
+        251..=254 => take(3_usize)(oi)?,
+        240..=249 => {
+            let o = &oi[..2];
+            (i, o)
+        },
+        250 => parse_ctl_suboption(i, oi)?,
+        _ => { panic!("unknown cmd {} => {:?}", cmd, oi); },
```
Comment (victorjulien) [inline rust/src/telnet/parser.rs:54]: consider how to handle this

File: src/app-layer-telnet.c [hunk 3]
```diff
@@ -0,0 +1,64 @@
+/* Copyright (C) 2018 Open Information Security Foundation
```
Comment (victorjulien) [inline src/app-layer-telnet.c:1]: get rid of these C files

**Comment by victorjulien** [comment]
I think maybe the SCLogNotice in one of the C files is causing the CIfuzz failures

**Comment by suricata-qa** [comment]
ERROR: QA failed on tlpr1_asan_suri.
  

Pipeline 5604

**Comment by victorjulien** [comment]
replaced by #6757

**Pull request closed without merging**